sys/src/9/port/sysfile.c

```diff
@@ -742,21 +741,21 @@ read(int fd, void *p, int32_t n, int64_t off)
 	 * The number of bytes read on this fd (c->offset) may be different
 	 * due to rewritings in mountfix.
 	 */
-		if(off == -1LL){	/* use and maintain channel's offset */
-			off = c->offset;
-		sequential = 1;
-	} else {
-		sequential = 0;
+	if(off == -1LL){	/* use and maintain channel's offset */
+		off = c->offset;
 	}
+	else
+		dopread = 0;
```

Comment:
Isn't this reversed? If off == -1LL, then it's NOT pread (we're relying on the offset tracked in the Chan). If it's pread, then we use the offset passed as an argument.